src/AutoRest.CSharp/Common/Input/Configuration.cs

```diff
@@ -264,6 +270,11 @@ internal static (string AbsoluteProjectFolder, string RelativeProjectFolder) Par
         public static bool SkipSerializationFormatXml { get; private set; }
         public static bool DisablePaginationTopRenaming { get; private set; }
 
+        public static bool AssertModelConstructorParametersForReadOnlyProperties { get; private set; }
+        public static bool PublicRestClients { get; private set; }
```

Comment:
I am wondering if we don't do this and simple add custom code to make it internal in the HLC libraries that will need this.

Proportionally this feels like a tiny % of the total libraires we have will want this.